### PR TITLE
Make Conway-specific queries only available in Conway

### LINKED ISSUE #4010: Make Conway specific queries only available in Conway
Consensus now makes it possible to create era specific queries: https://github.com/IntersectMBO/ouroboros-consensus/pull/922

We need to remove these three functions from `EraGov`:

* getDRepDistr
* getCommitteeMembers
* getConstitution

And move `Constitution` type into `Cardano.Ledger.Conway.Governance`

We need to also implement current queries in terms of Conway functionality only without worrying about previous eras.

### PR DESCRIPTION
**Pull request by aniketd**
# Description

Resolves #4010 

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/babbage/impl/cardano-ledger-babbage.cabal [hunk 1]
```diff
@@ -70,7 +70,7 @@ library
         cardano-ledger-binary ^>=1.3,
         cardano-ledger-core ^>=1.11,
         cardano-ledger-mary ^>=1.5,
-        cardano-ledger-shelley ^>=1.10,
+        cardano-ledger-shelley ^>=1.11,
```
Comment (aniketd) [inline eras/babbage/impl/cardano-ledger-babbage.cabal]: I hope I am understanding the PVP bumps needed correctly...

File: eras/shelley/impl/cardano-ledger-shelley.cabal [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cabal-version:      3.0
 name:               cardano-ledger-shelley
-version:            1.10.0.0
```
Comment (teodanciu) [inline eras/shelley/impl/cardano-ledger-shelley.cabal]: Version 1.10.0.0 of shelley is not released, so you don't need to bump the version to 1.11.0.0.0

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs [hunk 3]
```diff
@@ -674,11 +666,22 @@ class EraGov era => ConwayEraGov era where
   drepPulsingStateGovStateL :: Lens' (GovState era) (DRepPulsingState era)
   committeeGovStateL :: Lens' (GovState era) (StrictMaybe (Committee era))
 
+  getDRepDistr :: GovState era -> Map (DRep (EraCrypto era)) (CompactForm Coin)
+  getConstitution :: GovState era -> Constitution era
```
Comment (teodanciu) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs]: Why keep these functions here, `getConstitution` and `getCommitteeMembers` ?  can't we use the lenses that already exist in this class, `constitutionGovStateL` and `committeeGovStateL` ?    (and probably the same for  `getDRepDistr`) 
Or is there some restriction about not using lenses in the queries where they are used?

File: eras/conway/impl/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.13.0.0
 
+* Add `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `ConwayEraGov` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
```
Comment (teodanciu) [inline eras/conway/impl/CHANGELOG.md:6]: Could you `shift-tab` on this?  it's a little misaligned! Or is it on purpose?

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Governance.hs [hunk 5]
```diff
@@ -93,14 +79,6 @@ class
   getProposedPPUpdates :: GovState era -> Maybe (ProposedPPUpdates era)
   getProposedPPUpdates _ = Nothing
 
-  -- | Returns `Nothing` for all era preceding Conway, otherwise returns the hash of the constitution
-  getConstitution :: GovState era -> Maybe (Constitution era)
-  getConstitution = const Nothing
-
-  getCommitteeMembers ::
-    GovState era -> Maybe (Map (Credential 'ColdCommitteeRole (EraCrypto era)) EpochNo, UnitInterval)
-  getCommitteeMembers = const Nothing
```
Comment (teodanciu) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/Governance.hs:102]: How nice to get rid of this ugly things :grin:

**Review by teodanciu — COMMENTED** [review]
Depending on when you merge this, but to me it doesn't seem like you need any (EDIT:) **major** version bumps, because the backwards incompatible changes (removing the functions from the typeclasses)  are in shelley and can go in the unreleased 1.10 version, which the other packages already depend on.  Conway 1.13 which now contains  Constitution is also not released yet, so no bump needed there either. 
~If I understood correctly the slack conversation about versions and typeclass changes, unless any of the packages are exporting these removed functions, they don't need a bump.~ EDIT:  they do need non-breaking bump, sorry for the confusion

File: eras/shelley/impl/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,10 @@
 # Version history for `cardano-ledger-shelley`
 
+## 1.11.0.0
```
Comment (lehins) [inline eras/shelley/impl/CHANGELOG.md]: Looks like you never read the thing in bold in the PR where you place a checkmark :grin: 
> All visible changes are prepended to the latest section of a CHANGELOG.md for the affected packages. New section is never added with the code changes. 

In particular this one: **New section is never added with the code changes.** 

Version 1.10.0.0 has not been released yet

File: eras/allegra/impl/cardano-ledger-allegra.cabal [hunk 7]
```diff
@@ -1,6 +1,6 @@
 cabal-version:      3.0
 name:               cardano-ledger-allegra
-version:            1.3.0.1
+version:            1.3.1.1
```
Comment (lehins) [inline eras/allegra/impl/cardano-ledger-allegra.cabal]:
```suggestion
version:            1.3.1.0
```

File: eras/alonzo/impl/cardano-ledger-alonzo.cabal [hunk 8]
```diff
@@ -1,6 +1,6 @@
 cabal-version:      3.0
 name:               cardano-ledger-alonzo
-version:            1.6.0.1
+version:            1.6.1.1
```
Comment (lehins) [inline eras/alonzo/impl/cardano-ledger-alonzo.cabal]: You also need to change the version in the CHANGELOG as well
```suggestion
version:            1.6.1.0
```

File: eras/conway/impl/CHANGELOG.md (hunk 4 above)
Comment (aniketd) [reply eras/conway/impl/CHANGELOG.md:6]: This is on purpose. I use a single top-level item to mention the PR in which the change was introduced and sub-items to list out other changes.

File: eras/shelley/impl/CHANGELOG.md (hunk 6 above)
Comment (aniketd) [reply eras/shelley/impl/CHANGELOG.md]: Yes, sorry! My bad. I will revert all the bumps :sweat:

File: eras/conway/impl/CHANGELOG.md (hunk 4 above)
Comment (teodanciu) [reply eras/conway/impl/CHANGELOG.md:6]: Oh I see!  Thanks :+1:

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs (hunk 3 above)
Comment (aniketd) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs]: That's an interesting question. I think I agree with you. :+1:

**Comment by lehins** [comment]
>  they don't need a bump.

@teodanciu They do need a bump, just not a breaking version bump

**Comment by teodanciu** [comment]
> > they don't need a bump.
> 
> @teodanciu They do need a bump, just not a breaking version bump

Makes sense if I think about it. Thanks for clarifying

File: eras/shelley/test-suite/cardano-ledger-shelley-test.cabal [hunk 9]
```diff
@@ -1,6 +1,6 @@
 cabal-version:      3.0
 name:               cardano-ledger-shelley-test
-version:            1.3.0.2
+version:            1.3.1.0
```
Comment (lehins) [inline eras/shelley/test-suite/cardano-ledger-shelley-test.cabal]: There was no change to the package, why version bump?
```suggestion
version:            1.3.0.2
```

File: eras/shelley/test-suite/CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,10 @@
 # Version history for `cardano-ledger-shelley-test`
 
+## 1.3.1.0
+
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
+
```
Comment (lehins) [inline eras/shelley/test-suite/CHANGELOG.md]: Two things:
* Again: new section should never be added
* this is not a change that was applied to this package
```suggestion
```

File: eras/allegra/impl/CHANGELOG.md [hunk 11]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.3.1.0
 
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
```
Comment (lehins) [inline eras/allegra/impl/CHANGELOG.md]: This was the change in shelley, not in allegra
```suggestion
```

File: eras/alonzo/impl/CHANGELOG.md [hunk 12]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.7.0.0
 
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
```
Comment (lehins) [inline eras/alonzo/impl/CHANGELOG.md]: Ditto
```suggestion
```

File: eras/babbage/impl/CHANGELOG.md [hunk 13]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.7.0.0
 
+* Add the constructor `BabbageNonDisjointRefInputs` to `BabbageUtxoPredFailure`
+  * Utxo rule raises that `PredicateFailure` in Conway and future Eras when they are not disjoint.
```
Comment (lehins) [inline eras/babbage/impl/CHANGELOG.md]: You probably messed this up during a rebase. :grin: 
These same entries are at the bottom
```suggestion
```

File: eras/mary/impl/CHANGELOG.md [hunk 14]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.5.1.0
 
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
```
Comment (lehins) [inline eras/mary/impl/CHANGELOG.md]: Ditto
```suggestion
```

File: eras/shelley-ma/test-suite/CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,10 @@
 # Version history for `cardano-ledger-shelley-ma-test`
 
+## 1.2.2.0
+
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
+
```
Comment (lehins) [inline eras/shelley-ma/test-suite/CHANGELOG.md]: Ditto
```suggestion
```

File: eras/shelley-ma/test-suite/cardano-ledger-shelley-ma-test.cabal [hunk 16]
```diff
@@ -1,6 +1,6 @@
 cabal-version: 3.0
 name:          cardano-ledger-shelley-ma-test
-version:       1.2.1.7
+version:       1.2.2.0
```
Comment (lehins) [inline eras/shelley-ma/test-suite/cardano-ledger-shelley-ma-test.cabal]: There ar eno changes to the package
```suggestion
version:       1.2.1.7
```

File: libs/cardano-ledger-api/CHANGELOG.md [hunk 17]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.9.0.0
 
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
+  * Move `Constitution` to `Conway.Governance.Procedures`
```
Comment (lehins) [inline libs/cardano-ledger-api/CHANGELOG.md]: This change is not relevant for this package
```suggestion
```

File: libs/cardano-ledger-api/CHANGELOG.md [hunk 18]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.9.0.0
 
+* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` in `Shelley` #4033
```
Comment (lehins) [inline libs/cardano-ledger-api/CHANGELOG.md]:
```suggestion
* Remove `getDRepDistr`, `getConstitution` and `getCommitteeMembers` from `EraGov` #4033
```

**Review by lehins — CHANGES_REQUESTED** [review]
This looks good. Just need to fixup versions and changelogs

**Review by lehins — APPROVED** [review]
Looks great! Thank you.

**Pull request merged**